flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java

```diff
@@ -120,15 +146,55 @@ public void testAbortIgnoresMissing() throws Exception {
         runWithSyncWorker(this::callAbort);
     }
 
+    @Test
+    public void testCheckpointAbortBeforeStart() throws Exception {
+        SyncChannelStateWriteRequestExecutor worker = new SyncChannelStateWriteRequestExecutor();
+        ChannelStateWriterImpl writer = new ChannelStateWriterImpl(TASK_NAME, worker);
+        writer.open();
+        callAbort(writer);
+
+        // It should be started.
```

Comment:
nit: this comment is a bit miss leading. `it should be started` -> `start after abort should be ignored`?